pkg/geo/geomfn/node.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/twpayne/go-geom"
+	"github.com/twpayne/go-geom/xy"
+)
+
+// extractUniqueEndpoints extracts the endpoints from geometry and removes duplicates
+func extractUniqueEndpoints(g geo.Geometry) (geo.Geometry, error) {
+	endpoints, err := extractEndpoints(g)
+	if err != nil {
+		return geo.Geometry{}, fmt.Errorf("error extracting endpoints: %v", err)
+	}
+
+	output, err := UnaryUnion(endpoints)
```

Comment:
why do we call UnaryUnion?

Reply:
It was supposed to remove duplicated points. I have just realised that it's not needed.

Reply:
i think it still is.

Reply:
I mean, later in the code, it's going through each endpoint for each line, it breaks the loop once it finds any point for a particular line - this way removing `UnaryUnion` here should not affect the results, but it may affect the performance. I have added it here because it's used in PostGIS ([link](http://postgis.net/docs/doxygen/2.4/dd/d8f/lwgeom__geos__node_8c_a0ae5516a2dcf5d724c4b1652433076e5.html#a0ae5516a2dcf5d724c4b1652433076e5)). Should I keep it?

Reply:
If we check it later already then let's not do it here. Is there a test for this case?

Reply:
Yes, there is https://github.com/cockroachdb/cockroach/pull/56183/files#diff-f38103c4687b9c33fd756280757bbcafc674606264c9b5fdabc6b48b2868a7d6R484-R489, but for a duplicated LineString. Let me add one more for a LineString with duplicated points.